app/models/book.rb

```diff
@@ -1,3 +1,60 @@
 class Book < ApplicationRecord
   has_and_belongs_to_many :tags
+  has_one :reservation, dependent: :destroy
+  belongs_to :user, foreign_key: "user_id"
+
+  validate :cannot_edit_reserved_or_booked
+  before_destroy :cannot_destroy_reserved_or_booked
+
+  STATUSES = [
+    AVAILABLE = 'available',
+    RESERVED = 'reserved',
+    BORROWED = 'borrowed'
+  ]
+
+  validates :status, :inclusion => { :in => STATUSES }, :allow_blank => false
+
+  def belongs_to?(current_user)
+  	user == current_user
+  end
+
+  def is_available?
+    status == AVAILABLE
+  end
+
+  def available_to_borrow?(current_user)
+  	status == AVAILABLE && !belongs_to?(current_user)
```

Comment:
same as above, indentation and param name

Reply:
the name of this function is also a bit confusing, I'd suggest something like `available_to_borrow_by?(user)`

Reply:
Fixed 👍